### PR TITLE
added noise to pressure measurement

### PR DESCRIPTION
**Pull request by ptrkfry**
I noticed that even though you can defined a variance for the pressure sensor. The measurement was not noisy. I fixed that by adding normally distributed noise as it was done for GPS velocity for example.
One problem is now, that for variance = 0 the noise is very small, but not exactly 0. Not sure why that is.. Maybe it would make sense to include an exception for variance = 0 where no noise is added?

### CODE REVIEW

**Comment by hitimo** [comment]
add to whitelist

**Comment by hitimo** [comment]
@ptrkfry Could you please post a plot/data for 

> One problem is now, that for variance = 0 the noise is very small, but not exactly

**Comment by ptrkfry** [comment]
There was actually something missing. But the problem is still a nonzero noise for variance = 0. See plot:
pressure_noise_var_0
For higher variances it seems to make sense. See plot for variance = 10:
pressure_noise_var_10

File: rotors_gazebo_plugins/include/rotors_gazebo_plugins/gazebo_pressure_plugin.h [hunk 1]
```diff
@@ -17,6 +17,9 @@
 #ifndef ROTORS_GAZEBO_PLUGINS_PRESSURE_PLUGIN_H
 #define ROTORS_GAZEBO_PLUGINS_PRESSURE_PLUGIN_H
 
+// SYSTEM
```
Comment (hitimo) [inline rotors_gazebo_plugins/include/rotors_gazebo_plugins/gazebo_pressure_plugin.h]: rm comment (this should be obvious)

File: rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp [hunk 2]
```diff
@@ -76,6 +76,9 @@ void GazeboPressurePlugin::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf)
   getSdfParam<double>(_sdf, "referenceAltitude", ref_alt_, kDefaultRefAlt);
   getSdfParam<double>(_sdf, "pressureVariance", pressure_var_, kDefaultPressureVar);
 
+  // Initialize the normal distribution for pressure.
+  pressure_n_[0] = NormalDistribution(0, pressure_var_);
```
Comment (hitimo) [inline rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp]: std::normal_distribution takes mean and standard dev. as arguments (so sqrt(pressure_var_)

File: rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp [hunk 3]
```diff
@@ -117,6 +120,11 @@ void GazeboPressurePlugin::OnUpdate(const common::UpdateInfo& _info) {
       kPressureOneAtmospherePascals * exp(kAirConstantDimensionless *
           log(kSeaLevelTempKelvin / temperature_at_altitude_kelvin));
 
+  // Add noise to pressure measurement.
+  if(pressure_var_ != 0) {
```
Comment (hitimo) [inline rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp]: `if (pressure_var_ > 0.0) {`

File: rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp [hunk 4]
```diff
@@ -76,6 +76,9 @@ void GazeboPressurePlugin::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf)
   getSdfParam<double>(_sdf, "referenceAltitude", ref_alt_, kDefaultRefAlt);
   getSdfParam<double>(_sdf, "pressureVariance", pressure_var_, kDefaultPressureVar);
 
+  // Initialize the normal distribution for pressure.
```
Comment (hitimo) [inline rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp]: `CHECK(pressure_var_ > 0.0);`

File: rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp (hunk 2 above)
Comment (hitimo) [inline rotors_gazebo_plugins/src/gazebo_pressure_plugin.cpp]: Define magic parameter `0`:
```
double mean = 0.0;
pressure_n_[0] = NormalDistribution(mean, sigma);
```

File: rotors_gazebo_plugins/include/rotors_gazebo_plugins/gazebo_pressure_plugin.h [hunk 5]
```diff
@@ -17,9 +17,10 @@
 #ifndef ROTORS_GAZEBO_PLUGINS_PRESSURE_PLUGIN_H
 #define ROTORS_GAZEBO_PLUGINS_PRESSURE_PLUGIN_H
 
-// SYSTEM
 #include <random>
 
+#include <glog/logging.h>
```
Comment (hitimo) [inline rotors_gazebo_plugins/include/rotors_gazebo_plugins/gazebo_pressure_plugin.h]: `/home/jenkins/workspace/rotors_simulator/osx/ubuntu-xenial/src/rotors_simulator/rotors_gazebo_plugins/include/rotors_gazebo_plugins/gazebo_pressure_plugin.h:22:26: fatal error: glog/logging.h: No such file or directory
compilation terminated.`

**Comment by ptrkfry** [comment]
Hm, now it runs for the ubuntu configuration, but fails on osx. Do you see why that is?
Also, does it make sense at all to include the glog_catkin dependency? Can people outside of ETH use this too?

**Comment by hitimo** [comment]
test this please

File: rotors_gazebo_plugins/package.xml [hunk 6]
```diff
@@ -36,6 +36,7 @@
   <build_depend>tf</build_depend>
   <build_depend>yaml-cpp</build_depend>
   <build_depend>protobuf-dev</build_depend>
+  <build_depend>glog_catkin</build_depend>
```
Comment (hitimo) [inline rotors_gazebo_plugins/package.xml]: Alphabetize

File: rotors_gazebo_plugins/package.xml [hunk 7]
```diff
@@ -53,6 +54,6 @@
   <run_depend>std_srvs</run_depend>
   <run_depend>tf</run_depend>
   <run_depend>yaml-cpp</run_depend>
-
+  <run_depend>glog_catkin</run_depend>
```
Comment (hitimo) [inline rotors_gazebo_plugins/package.xml]: Alphabetize

**Pull request merged**